receiver/zookeeperreceiver/scraper.go

```diff
@@ -159,22 +160,11 @@ func (z *zookeeperMetricsScraper) appendMetrics(scanner *bufio.Scanner, rms pdat
 				)
 				continue
 			}
-			metric := ilm.Metrics().AppendEmpty()
-			initMetric(metric)
-			switch metric.DataType() {
-			case pdata.MetricDataTypeGauge:
-				dp := metric.Gauge().DataPoints().AppendEmpty()
-				dp.SetTimestamp(now)
-				dp.SetIntVal(int64Val)
-			case pdata.MetricDataTypeSum:
-				dp := metric.Sum().DataPoints().AppendEmpty()
-				dp.SetTimestamp(now)
-				dp.SetIntVal(int64Val)
-			}
-			keepRM = true
+			recordDataPoints(now, int64Val)
 		}
 	}
-	if keepRM {
+	z.mb.Emit(ilm.Metrics())
+	if ilm.Metrics().Len() > 0 {
```

Comment:
Can you help me understand why this block is necessary / what it does?

Reply:
Just to make sure that we don't send `ResourceMetrics` down the pipeline if no metrics were recorded. This is just a replacement for the `keepRM` variable